### ISSUE TITLE
oatpp::String / oatpp::base::StrBuffer interoperable with std::string

### ISSUE THREAD

**Issue by grisuthedragon** [oatpp/oatpp#470]
Is there a way to get an easy way in both directions between `std::string` and Oat++'s String/StrBuffer types. Using 
```
 str->std_str() 
```
or 
```
 String::createShared(sstr.c_str()) 
``` 
at every point creates a lot of bloat in the code when dealing with other libraries, that relying on the `std::string` type. 

If there is no way, one can see this as feature request. I would be find if 
```
std::string sstr("test");
auto ostr = oatpp::String::createShared(sstr); 
``` 
and 
```
auto ostr = oatpp:String("mystring");
std::string sstr = ostring;
```
would work. That you make huge parts of our code better readable.

**Comment by bamkrs** [comment]
Hey Martin,

that's one of the most common Problems with Oat++ strings, not gonna lie 👀
Have you checked the v1.3.0 branch? It's our next release. One of its mayor changes include the refactoring `oatpp::String` to be 'just a wrapper' around `std::string`. 
This highly reduces the bloat and prevents unnecessary copying of data.

**Comment by grisuthedragon** [comment]
Basically this looks good but having the `*` in one direction is a bit strange. So having the proper type cast function in this place would be great. Another problem is, that this breaks the swagger support at the moment.

**Comment by bamkrs** [comment]
Hey Martin,

you are welcome to contribute a pull request, we always encourage our community to provide their own PR's so its openly documented where the suggestions comes from :)
If you would rather not, we take care.

Also, swagger does have an `1.3.0` branch, too. Is it broken? Haven't checked in a while and `1.3.0` introduces some major refactoring of core components.

**Comment by grisuthedragon** [comment]
I added the patch, including the requirements for the `oatpp-swagger` module.  The a separate PR https://github.com/oatpp/oatpp-swagger/pull/50 is placed to let it work again.

**Comment by lganzzzo** [comment]
Fixed in `1.3.0` - see [the new oatpp::String](https://github.com/oatpp/oatpp/blob/master/changelog/1.3.0.md#the-new-oatppstring)
